scopes/react/component-highlighter/frame/frame.tsx

```diff
@@ -1,64 +1,98 @@
-
-const popperModifiers: Modifier<any>[] = [
-  ignorePopperSize,
-  resizeToMatchReference,
-  {
-    name: 'flip',
-    enabled: false,
-  },
-  {
-    name: 'offset',
-    options: {
-      // move box from above the target ('top-start')
-      // to directly cover the target.
-      offset: ({ reference }: any) => [BASE_OFFSET, BASE_OFFSET - reference.height],
-    },
-  },
-];
+/** frame padding around the target */
+const MARGIN_FROM_TARGET = +classStyles.offset;
+/** min. distance from the edge of the screen. Kept for syncing */
+const MARGIN_FROM_DOC_EDGE = 0;
 
 export interface FrameProps extends React.HTMLAttributes<HTMLDivElement> {
-  targetRef: HTMLElement | null;
+  targetElement: HTMLElement | null;
   stylesClass?: string;
   /** continually update frame position to match moving elements */
   watchMotion?: boolean;
 }
 
+// position - bottom start (bottom left corner)
+// x - width - horizontal (cross axis)
+// y - height - vertical (main axis)
+
 export function Frame({
-  targetRef,
+  targetElement,
   watchMotion,
   className,
   stylesClass = classStyles.overlayBorder,
   style,
 }: FrameProps) {
-  const [referenceElement, setReferenceElement] = useState<HTMLDivElement | null>(null);
+  const dimensionRef = useRef({ width: 0, height: 0 });
+  const shiftRef = useRef<Coords | undefined>();
 
-  const { styles, attributes, update } = usePopper(targetRef, referenceElement, {
-    modifiers: popperModifiers,
-    placement: 'top-start',
+  const { x, y, strategy, reference, floating, update, refs } = useFloating({
+    placement: 'bottom-start',
+    middleware: [
+      // reposition x,y, to the top of the reference
+      offset((options) => -options.reference.height),
+      // offset the frame by its extra padding
+      offset(() => ({ mainAxis: -MARGIN_FROM_TARGET, crossAxis: -MARGIN_FROM_TARGET })),
+      // push the frame inside the screen
+      shift({ rootBoundary: 'document', padding: MARGIN_FROM_DOC_EDGE, mainAxis: true, crossAxis: true }),
+      // read "shift" for the size-apply() method (because it doesn't forward middlewareData)
+      {
+        name: 'read-shift',
+        fn({ middlewareData }) {
+          shiftRef.current = middlewareData.shift;
+          return {};
+        },
+      },
+      // size also applies overflow detection via width and height
+      size({
+        // apply overflow detection in reference to the document
+        rootBoundary: 'document',
+        padding: MARGIN_FROM_DOC_EDGE,
+        apply({ reference: referenceRect, height, width }) {
+          const paddingX = 2 * MARGIN_FROM_TARGET - (shiftRef.current?.x || 0);
+          const paddingY = 2 * MARGIN_FROM_TARGET - (shiftRef.current?.y || 0);
+
+          dimensionRef.current = {
+            width: Math.min(referenceRect.width + paddingX, width),
+            height: Math.min(referenceRect.height + paddingY, height),
+          };
+          Object.assign(refs.floating.current?.style, dimensionRef.current);
```

Comment:
yes, floating-ui suggests applying directly from the `apply` method, because it requires all coordinates to be recalculated.